### ISSUE TITLE
Translate into Japanese

### ISSUE THREAD

**Issue by zacky1972** [parcel-bundler/website#94]
We'll translate the parcel website into Japanese!

**Comment by corocn** [comment]
👍

**Comment by Pittan** [comment]
@zacky1972 Is there something I can help?

**Comment by imncls** [comment]
does anyone already started work for this?
i have already started work.

**Comment by Pittan** [comment]
@imncls I'm starting working on it 🙋

**Comment by Pittan** [comment]
@imncls  
Sorry, I mean I have a local translated file, but not pushed on any repositories.
It's too outdated and I'm working on to update it.

**Comment by imncls** [comment]
@Pittan please don't apologize.
but, duplicate work is simply a waste of time. so if you do work i should be quit my current work.
well, when you need help, i'll help you.

**Comment by Pittan** [comment]
@imncls Thank you 😭 
New Japanese resources will be ready in a few days 🙏

**Comment by imncls** [comment]
@Pittan np~
that's really cool.
imo parcel website has many problems now, so if you sent a pr i will make a new pr.

**Comment by dandv** [comment]
Why do you guys think it's a good idea to translate programming documentation into languages other than English?

http://blog.codinghorror.com/the-ugly-american-programmer/

https://wiki.dandascalescu.com/essays/english-universal-language

**Comment by watilde** [comment]
@dandv Here are some reasons why translation is v good idea for tech communities. Not sure for this specific case, but generally.

1. People cannot choose their mother tongue.
2. English learning differs in difficulty depending on mother tongues.
3. It will be an even more significant barrier for kids.
4. Google translate between Japanese and English does not work well sometimes.

Looking at W3C's mission, they also mentions `regardless of native language`:
> One of W3C's primary goals is to make these benefits available to all people, whatever their hardware, software, network infrastructure, native language, culture, geographical location, or physical or mental ability.

If we want to have more people who have different backgrounds and believe it's a good idea, we should get rid of the barrier.

Refs:
+ [Language Learning Difficulty](http://aboutworldlanguages.com/language-difficulty)
+ [Web for All
](https://www.w3.org/Consortium/mission)

**Comment by dandv** [comment]
> If we want to have more people who have different backgrounds and believe it's a good idea, we should get rid of the barrier.

Which would be a more effective way of getting rid of the language barrier? Translating all materials in all languages, or encouraging people to use one universal language?

**Comment by watilde** [comment]
> Translating all materials in all languages, or encouraging people to use one universal language?
For me, the two options turned into: some translators make efforts vs all human learns English. The former is more efficient.

I know it's p challenging to maintain the translated docs as a core maintainer, but some solutions are already existed and used in many popular projects. I see that helps new joiners a lot :)

**Comment by ymotongpoo** [comment]
> Which would be a more effective way of getting rid of the language barrier? Translating all materials in all languages, or encouraging people to use one universal language?

I don't think these 2 options are exclusive each other, and I believe it is good thing that the community is willing to put efforts on the OSS project in any form, say localization. Even if the committers or contributors for this repository decide to reject further contribution on l10n here, still they may create a dedicated repo for it. I think it is preferred to have all materials in one central repo including l10n.

I understand that it is ideal for all developers to understand English, but making it real just by encouraging developers to learn English takes long time, and it can not be achieved only by us; it involves English education in schools. Also, in terms of English education, the translated doc would be one of the best learning materials for English self study for the users of this tool.

**Comment by dandv** [comment]
> For me, the two options turned into: some translators make efforts vs all human learns English. The former is more efficient.

No. If some translators make some effort, then some materials will be available in some languages. If everyone learns and uses English, everyone understands all materials. We have a similar problem right now with plugs and outlets: some manufacturers make converters from some plug types to some outlets, but if everyone used one standard, things would be much easier. We've achieved that type of success with USB-C.

![image](https://user-images.githubusercontent.com/33569/52606656-41432600-2e28-11e9-8fa9-d685c52486ff.png)

> Also, in terms of English education, the translated doc would be one of the best learning materials for English self study for the users of this tool.

How exactly would a translated doc be even a good learning material for English self study? Do you think users would bother to read the translated and the English doc side by side?

**Comment by Pittan** [comment]
Sorry for interrupt. Could I challenge the translation again?  
I love how parcel works for small front-end development. 😭

**Comment by uetchy** [comment]
It's great to have translated documents in OSS because there are a lot of people eager to use and contribute to OSS however struggling for language and wasting a great deal of time. That's a needless loss for non-English speakers.
I can argue that there is no need to translate docs If English is the truly universal language. But not at all.
Please don't think localization stuff is waste of time but rather think it will cut a large amount of time of many.

**Comment by ymotongpoo** [comment]
> How exactly would a translated doc be even a good learning material for English self study? Do you think users would bother to read the translated and the English doc side by side?

Yeah, I just raised an example, and is the situation problem? (Your reaction sounds so.)

**Comment by lulalala** [comment]
> We've achieved that type of success with USB-C.

Aha such a noble goal.  Unfortunately as much as we want language to work like a standard or program, in reality it does not work this way (otherwise Esperanto would be more wide spread now).

Only when we can refactor a language (e.g. make December the 10th month of the year, because you know... "decimal" means base-ten), would it be possible to require everyone to learn English.

**Comment by agektmr** [comment]
Learning something by yourself and letting someone learn something are different things. If you want to learn something and you don't know the language, go ahead and learn the language. But for letting other people learn things, providing translation is more efficient. It's obviously more efficient for a single person to translate a document comparing to 10,000 people learn the language by themselves.

At least that's the approach Developer Advocate should choose if you want to reach more people.

**Comment by nfang** [comment]
Translating documentation would be a more efficient way of getting people from different culture/language background to get involved in this particular project. Talking about getting rid of language barrier in general is way off the topic.

**Comment by cedmax** [comment]
Arguably the common language should be Mandarin Chinese: it's the language with the highest number of native speakers (and the trend is for the gap to increase).

Suggesting English to be universal is arrogant, self centered and utterly elitist. We need an inclusive approach, reaching out to communities and to enable diversification. I, for one, I'm sad this kind of bs always comes from a white dude, I wish we were better represented.

**Comment by dandv** [comment]
I completely agree that translating this particular documentation is more efficient at *getting more developers to use Parcel*.

That's a nice short-term goal. It's also just that - a narrow, short term, particular goal.

My goal is broader - empowering everyone in the world to access as much information as possible. Or in [the words of my former employer](https://about.google/), making information universally accessible. Google is going about that with machine translation. That's great. In parallel, we can go about it by encouraging people to learn English, which, [like it or not](https://wiki.dandascalescu.com/essays/english-universal-language#Where_we_are_today), is the universal language nowadays.

@cedmax: the white dude in question is [not a native English speaker](https://wiki.dandascalescu.com/essays/english-universal-language#Introduction). Please read that section and the rest from my article. I'm sad these kinds of weak counter-arguments come again and again. If there's a new argument I haven't covered, please comment there.

**Comment by jakearchibald** [comment]
Although @dandv was working for Google when he made his comment, his view is not shared by anyone else I know at devrel.

I feel incredibly lucky that documentation & standards are often written in English. I wouldn't be where I am today without those resources. Trying to prevent others having resources in their first language is absurd.

**Comment by dandv** [comment]
@jakearchibald I have the utmost respect for you, thanks for jumping in, and for nudging me to clarify my point:

Yes - it's fortunate that documentation & standards are often written in English; or rather, intelligent on the part of the authors. What would an author choose if they want to reach the widest audience - to publish documentation in language X (perhaps their own), or in the language with the largest number of developer, *and* largest number of bilingual speakers who could translate it into other languages? I hope we agree it's the latter choice. This means you're not lucky that documentation & standards are often written in English; you're lucky that you're a native English speaker and didn't have to learn English. I did have to. And it's been the best time investment I've made in my entire life.

That's what I'm trying to convey: feel free to translate all you want; but understand that by doing so, you're removing an incentive for those folks who read the translated material to learn English. And let's be realistic - they're already developer; they already know quite a bit of English because most of the documentation is written primarily in English (translations, if they exist, often lag behind). If you want to be a top developer, it's in your best interest to learn English.

**Comment by cedmax** [comment]
@dandv it doesn't quite matter what is your original language is, what matters is the cultural background in which your opinions got shaped. I appreciate you might (think you) have a very good reason for what you are suggesting. It doesn't make it less elitist. 

On the white dude note: that's not a counter argument, that's a statement about a trend I noticed. Sorry if I went off topic on this one

**Comment by esparta** [comment]
Being a English-native speaker and also say every body should (must?) learn English is a sad situation to be involved at, please reconsider what you are saying, @dandv.

As someone who has been helping others to make accessible technical documentation, [tutorials][git-immersion] & [books][97cosas] for no-native English speakers (Spanish, to be more precise) I can attest this help, more than you ever know being lucky and privileged enough to read and write the de-facto (current) language. Saying "I did it, you also can" would be just condescend and a clear case of 'survivor-bias'. I was privileged enough to attend public and private schools where I was able to learn English, I had time to learn on my own pace English when I needed, a lot of people didn't.

This situation is not new, the first books (and knowledge in general) were all done on the de-facto language, all of them. People were NOT able to access them: hebrew, greek and latin... So, everybody need to read on those languages, like it or not?

IMO, even with good intentions this is nothing different that the [Duke Professor Urging Chinese Students Not To Speak Chinese][duke], is for their own good. Good intentions are not enough, you need to live and be on their situations. 

[duke]: https://www.npr.org/sections/goatsandsoda/2019/01/29/689660523/duke-professor-is-sorry-for-urging-chinese-students-not-to-speak-chinese?t=1549958458100
[git-immersion]: https://github.com/esparta/gitimmersion-spanish
[97cosas]: https://97cosas.com

**Comment by dandv** [comment]
> Being a English-native speaker and also say every body should (must?) learn English is a sad situation to be involved at

Who are you talking about? Had you read the *Introduction* to [my essay](https://github.com/parcel-bundler/website/issues/94#issuecomment-447616587), you'd know that I'm not a native English speaker.

**Comment by esparta** [comment]
I read it, that's why I also wrote this:

> Saying "I did it, you also can" would be just condescend and a clear case of 'survivor-bias'. 

I'm not a native-speaker, but you and myself have the level of one of them... I didn't say you were born on a place or was your first language. Are you stopping exactly where you are not comfortable enough?

**Comment by fabiosussetto** [comment]
@dandv 

> That's what I'm trying to convey: feel free to translate all you want; but understand that by doing so, you're removing an incentive for those folks who read the translated material to learn English.

Your reasoning is just based on assumptions that you think hold true for everyone, without even questioning them. 

You can't decide what works as an incentive or not for other people on such a subjective topic. It may well be that having the same documentation both in English and Japanese is a *bigger* incentive to learn English OR Japanese, simply because you can read it in both your native language and in the language you're learning. 

In fact, I bet you learned English mostly because someone translated content you were interested in from Romanian to English.

**Comment by alicetragedy** [comment]
> My goal is broader - empowering everyone in the world to access as much information as possible.

Great, so if you really want to empower _everyone_ (including minorities and underrepresented people) please stop using the word “guys” when you're talking to a variety of people, as that term is exclusive and not gender-neutral.

**Comment by devongovett** [comment]
The goal of our documentation is to allow anyone to quickly and easily learn about Parcel, no matter their background or proficiency in any language. We don't wish to exclude anyone from using Parcel simply because they don't speak English. This goal is accomplished by providing the documentation in as many languages as possible.

English speakers are incredibly lucky that lots of documentation is available for libraries and tools in their language, and I do not see why it should be any different for speakers of other languages. The barrier to entry to learning a tool should not also include learning English. That's a crazy high barrier. Obviously things would be easier if everyone in the world just spoke the same language, but that is not a reality or even a remote possibility in any reasonable future. Machine translations will continue to get better, but for now don't match human translations in terms of quality. Therefore, we provide human translations which contributors have generously provided for others of a similar background to them.

Locking this issue to collaborators. If you wish to help translate the website into Japanese as the original issue intended, please carry on. Otherwise, this discussion can occur on other platforms.
